File: my-folder/0141-linked-list-cycle/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if not head:
            return False
        temp = head
        temp2 = head

        while temp.next is not None and temp2.next is not None and temp2.next.next is not None:
            if temp.next == temp2.next.next:
                return True
            temp = temp.next
            temp2 = temp2.next.next
        return False

File: my-folder/0141-linked-list-cycle/solution.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
        if (head == null) return false;
        ListNode n1 = head;
        ListNode n2 = head;
        while (n1.next != null && n2.next != null && n2.next.next != null) {
            n1 = n1.next;
            n2 = n2.next.next;
            if (n1 == n2) return true;
        }
        return false;
    }
}
